org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurConsole.java

```diff
@@ -375,9 +399,16 @@ private void do_pagedown() {
 	 * where>=0; append the text if where<0.
 	 */
 	private void do_add(int where, String text, AttributeSet style) {
-		StyledDocument doc = main.getStyledDocument();
 		try {
-			doc.insertString(where >= 0 ? where : doc.getLength(), text, style);
-		} catch (BadLocationException ex) {}
+			StyledDocument doc = main.getStyledDocument();
+			if (TableView.isTable(text)) {
+				Table table = TableView.toTable(text, false);
+				
+				doc.insertString(where >= 0 ? where : doc.getLength(),table.toString(), mono);
+				main.getCaret().setSelectionVisible(false);
+			} else
```

Comment:
add curlies around this `else` block (since the corresponding `then` block has curlies)